src/chainlink/oracle/ChainlinkOracleImpl.sol

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {AddressUtils} from "splits-utils/AddressUtils.sol";
+import {AggregatorV3Interface} from "chainlink/interfaces/AggregatorV3Interface.sol";
+import {ConvertedQuotePair, QuotePair, QuoteParams} from "splits-utils/LibQuotes.sol";
+import {QuotePair, QuoteParams} from "splits-utils/LibQuotes.sol";
+import {TokenUtils} from "splits-utils/TokenUtils.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {OracleImpl} from "../../OracleImpl.sol";
+import {ChainlinkPairDetails} from "../../libraries/ChainlinkPairDetails.sol";
+import {ChainlinkPath} from "../../libraries/ChainlinkPath.sol";
+
+/// @title Chainlink Oracle Implementation
+/// @author 0xSplits
+/// @notice A clone-implementation of an oracle using Chainlink
+/// @dev This contract uses token = address(0) to refer to ETH.
+contract ChainlinkOracleImpl is OracleImpl {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using AddressUtils for address;
+    using TokenUtils for address;
+    using ChainlinkPairDetails for mapping(address => mapping(address => PairDetail));
+    using FixedPointMathLib for uint256;
+    using ChainlinkPath for bytes;
+
+    /// -----------------------------------------------------------------------
+    /// errors
+    /// -----------------------------------------------------------------------
+
+    error InvalidPair_FeedNotSet(QuotePair qp);
+    error StalePrice(AggregatorV3Interface feed, uint256 timestamp);
+    error NegativePrice(AggregatorV3Interface feed, int256 price);
+
+    /// -----------------------------------------------------------------------
+    /// structs
+    /// -----------------------------------------------------------------------
+
+    struct InitParams {
+        /// @notice owner of the contract
+        address owner;
+        /// @notice paused state of the contract
+        bool paused;
+        /// @notice initial pair details
+        SetPairDetailParams[] pairDetails;
+    }
+
+    struct SetPairDetailParams {
+        QuotePair quotePair;
+        PairDetail pairDetail;
+    }
+
+    struct PairDetail {
+        /// @notice packed encoded feed[]
```

Comment:
is manually unpacking calldata fine for 1/ the subgraph and 2/ goldsky's sql mirror pipeline?

Reply:
(i imagine so just think we should double check)

Reply:
yeah it should be